platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/EncryptingWrappedPersistenceManager.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.felix.cm.file;
+
+import static org.osgi.framework.Constants.SERVICE_PID;
+import static org.osgi.service.cm.ConfigurationAdmin.SERVICE_FACTORYPID;
+
+import java.io.IOException;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.Set;
+
+import org.apache.felix.cm.PersistenceManager;
+
+import ddf.security.encryption.EncryptionService;
+import ddf.security.encryption.impl.EncryptionServiceImpl;
+
+/**
+ * Performs encryption.
+ */
+public class EncryptingWrappedPersistenceManager extends WrappedPersistenceManager {
+
+    private static final Set<String> EXCLUDED_PROPERTIES = new HashSet<>();
+
+    static {
+        EXCLUDED_PROPERTIES.add(SERVICE_PID);
+        EXCLUDED_PROPERTIES.add(SERVICE_FACTORYPID);
+        // Not worth adding a dependency on felix's fileinstall module just for the constant
+        EXCLUDED_PROPERTIES.add("felix.fileinstall.filename");
+    }
+
+    private final EncryptionService encryptionService;
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager) {
+        super(persistenceManager);
+        this.encryptionService = new EncryptionServiceImpl();
+    }
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager,
```

Comment:
❓ Is this constructor only for testing? If so then let's make it package private and annotated with @VisibleForTesting